pkgs/checks/doc/migrating_from_matcher.md

```diff
@@ -66,6 +66,12 @@ check(actualCollection).deepEquals(expected);
     value to check for equality, a matcher for the emitted value, or a matcher
     for the entire queue which would match multiple values. Use `it()..emits()`
     to check the emitted elements.
+-   The [`matches` Matcher][matches] would convert a `String` argument into a
+    `Regex`, so `matches(r'\d')` would match the value `'1'`. With
+    `matchesPattern` a `String` argument is treated as a `Pattern` instead of
```

Comment:
A bit confused by this description, it sounds like we are converting a `String` to a `Pattern` or something -if you don't realize that `String` already implements pattern then it isn't obvious that we are just leaving the string alone.

Reply:
I expanded this. Is it more clear now?